### DISCUSSION TITLE
Why eventStore do not use plant event table?

### DISCUSSION THREAD

**Discussion by ChopperLee2011** [looplab/eventhorizon#356] [Q&A] [answered]
As I see, Eventhorizon store the struct ( maybe I can call it aggregate table ) in the eventStore, likes: `{aggregateId: string, events: [] object}`, not the simple event table,like `{aggregateId: string, event: object }`. 
This makes some events need to update the record of the eventStore, and I think this is not safe with race condition. 
Why not just do add operation for the eventStore?
Maybe I am wrong. Appreciate for read and answer my question.

**Comment by maxekman** [comment]
There could definitely exist race conditions. Are you referring to the MongoDB implementation of the event store? If so could you provide a short example of how the add operation could be done without removing the atomicity of the increment?

**Comment by maxekman** [comment]
I assume you are interested in these lines: https://github.com/looplab/eventhorizon/blob/master/storage/mongodb/eventstore.go#L157

**Comment by maxekman** [comment, marked as answer]
I'll close this with the conclusion that Event Horizon is not yet thread safe.
